omod-common/src/main/java/org/openmrs/module/webservices/helper/TaskServiceWrapper.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.webservices.helper;
+
+import org.openmrs.scheduler.TaskDefinition;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.api.context.Context;
+
+import java.util.Collection;
+
+/**
+ * It will used to retrieve the required data from the Scheduler Service
+ */
+public class TaskServiceWrapper {
```

Comment:
What value does this class add?

Reply:
Wrapping the required task information between related resource and SchedulerService

Reply:
This wrapping is simply one method call hence rendering it useless.

Reply:
yah, but in the flow I want this class to create another Mock class for testing. 
Otherwise, I can't do the testing. 
I am using **TaskServiceWrapper** for active service works 
**MockTaskServiceWrapper** for mock testing works

Reply:
You can always mock Context.getSchedulerService().getTask(), etc without writing new mock classes.

Reply:
Here I followed the same way of the ModuleActionResource and ModuleWrapper and achieve the target :blush: 
What do you mean by the mocking Services?

Reply:
Ok, let us go with what you have. 😊